### PR TITLE
enable TLS if possible

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5420

enable TLS if possible and force use of TLSv1

### CODE REVIEW

**Comment by stephdl** [comment]
we could add (from http://www.postfix.org/TLS_README.html#server_cipher)
 
```
    smtpd_tls_mandatory_ciphers = high
    smtpd_tls_mandatory_exclude_ciphers = aNULL, MD5
```

but I worry about compatibility

File: root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# SMTPD
+smtpd_tls_security_level = may
+smtpd_tls_auth_only = no
+smtpd_tls_key_file = /etc/postfix/postfix.key
+smtpd_tls_cert_file = /etc/postfix/postfix.crt
+smtpd_tls_received_header = yes
+smtpd_tls_loglevel = 1
+smtpd_tls_session_cache_timeout = 3600s
+tls_random_source = dev:/dev/urandom
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS]: Do not set options, if the value is already provided by default (i.e. see `postconf -d | grep tls_random_source`)

File: root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# SMTP
+smtp_tls_security_level = may
+smtp_tls_loglevel = 1
+smtp_tls_cert_file = /etc/postfix/postfix.crt
+smtp_tls_key_file = /etc/postfix/postfix.key
+
+# Force TLSv1
+smtpd_tls_mandatory_protocols=!SSLv2,!SSLv3
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS]: Actual value should be controlled by a centralized prop. We should define that prop, for instance `tls/policy` - allowed values:

* `legacy` the current default settings, before Mar 14 2018
* `20180314`, the new policy we're implementing
* future policies identified by release date

Each package is responsible to implement the policy in its templates, by providing the actual settings for OpenSSL (or any other TLS library). Unknown values should be mapped to the latest (strongest) defined policy.
See also https://community.nethserver.org/t/gdpr-and-ssl-hardening/9006/17

File: root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS [hunk 3]
```diff
@@ -0,0 +1,27 @@
+
+#
+# TLS
+#
+
+# SMTPD
+smtpd_tls_security_level = may
+smtpd_tls_auth_only = no
+smtpd_tls_key_file = /etc/postfix/postfix.key
+smtpd_tls_cert_file = /etc/postfix/postfix.crt
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/postfix/main.cf/40enableTLS]: SSL certificates are currently set when mail-server is installed. Postfix relies on dovecot certs for SMTP/AUTH over TLS `smtpd_tls_auth_only = yes`.

See also http://www.postfix.org/TLS_README.html#server_tls_auth

As Postfix now requires a certificate also for sending messages (i.e. in a gateway-only use case),  we must decide how to split the new configuration, and we must do in a way that is compatible with both mail and mail2 implementations.

**Comment by DavidePrincipi** [comment]
We might better design this implementation if we solve this issue first

https://github.com/NethServer/dev/issues/5421

**Comment by stephdl** [comment]
yes let's concentrate on NethServer/dev#5421 first, I will propose a list of service

**Comment by stephdl** [comment]
@DavidePrincipi can you comment on this commit to enable tls for smtp

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**